website/payments/api/v2/admin/views.py

```diff
@@ -0,0 +1,154 @@
+import rest_framework.filters as framework_filters
+from django.apps import apps
+from django.http import Http404
+from django.utils.translation import gettext_lazy as _
+from oauth2_provider.contrib.rest_framework import IsAuthenticatedOrTokenHasScope
+from rest_framework import status, serializers
+from rest_framework.exceptions import PermissionDenied, ValidationError
+from rest_framework.generics import get_object_or_404, GenericAPIView
+from rest_framework.response import Response
+from rest_framework.settings import api_settings
+
+from payments import services, payables, NotRegistered
+from payments.api.v2 import filters
+from payments.api.v2.admin.serializers.payable_create import PayableCreateSerializer
+from payments.api.v2.admin.serializers.payable_detail import PayableSerializer
+from payments.api.v2.admin.serializers.payment import (
+    PaymentSerializer,
+    PaymentCreateSerializer,
+)
+from payments.exceptions import PaymentError
+from payments.models import Payment, PaymentUser
+from thaliawebsite.api.v2.admin import (
+    AdminListAPIView,
+    AdminCreateAPIView,
+    AdminRetrieveAPIView,
+    AdminDestroyAPIView,
+    AdminUpdateAPIView,
+    AdminPermissionsMixin,
+)
+
+
+class PaymentListCreateView(AdminListAPIView, AdminCreateAPIView):
+    """View that allows you to create and list payments as admin."""
+
+    queryset = Payment.objects.prefetch_related(
+        "paid_by__profile",
+        "paid_by__membership_set",
+        "processed_by__profile",
+        "processed_by__membership_set",
+    )
+
+    required_scopes = ["payments:admin"]
+    filter_backends = (
+        framework_filters.OrderingFilter,
+        filters.CreatedAtFilter,
+        filters.PaymentTypeFilter,
+    )
+    ordering_fields = ("created_at",)
+
+    def get_serializer_class(self):
+        if self.request.method.lower() == "post":
+            return PaymentCreateSerializer
+        return PaymentSerializer
+
+    def create(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        self.perform_create(serializer)
+        return Response(
+            PaymentSerializer(
+                serializer.instance, context=self.get_serializer_context()
+            ).data,
+            status=status.HTTP_201_CREATED,
+        )
+
+
+class PaymentDetailView(AdminRetrieveAPIView, AdminDestroyAPIView):
+    """View that allows you to manage a single payment as admin."""
+
+    queryset = Payment.objects.all()
+    serializer_class = PaymentSerializer
+    permission_classes = [IsAuthenticatedOrTokenHasScope]
+    required_scopes = ["payments:admin"]
+
+
+class PayableDetailView(AdminPermissionsMixin, GenericAPIView):
```

Comment:
I'm getting error
`'PayableDetailView' should either include a `queryset` attribute, or override the `get_queryset()` method.`